app/src/main/java/com/baijum/ukufretboard/data/sync/BackupRestoreManager.kt

```
@@ -534,6 +535,106 @@ class BackupRestoreManager(
         return json.encodeToString(JsonObject.serializer(), out)
     }
 
+    /**
+     * Normalizes timestamp fields in KMP-format backups that may contain
+     * fractional Doubles (from iOS JSONSerialization) or epoch-seconds values.
+     *
+     * Truncates fractional parts so kotlinx.serialization can decode to Long,
+     * and converts seconds to milliseconds using the same heuristic as
+     * [normalizeIosFavorites].
+     */
+    private fun normalizeKmpTimestamps(jsonContent: String): String {
```

detekt (1.23.8)

[warning] The function normalizeKmpTimestamps appears to be too complex based on Cognitive Complexity (complexity: 29). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)